### ISSUE TITLE
Font's emoji glyphs are squished despite settings

### ISSUE THREAD

**Issue by tom7** [mintty/mintty#1265]
Thanks so much for your work on mintty. It's great.

I have a font with (some) emoji glyphs. They are just regular outlines like the rest of the characters. It's a monospaced font, so the emojis are the same width as everything else. I'd like if these glyphs rendered at the font's indicated size and scale, like ASCII does. (I made the font; it's [here](http://tom7.org/fixedersys/FixederSys2x.ttf)).

By default, these emoji display like this in mintty:

![bomb-wide](https://github.com/mintty/mintty/assets/499244/e67f3af6-a9eb-47b6-a5ad-6dbba123ea6c)

The bomb emoji here is output as `" \U0001F4A3 "` from my program, so I'd like it to occupy the same space as `"   "` on the other lines.

Mintty is rendering this as a double-width character, which breaks the alignment of this text. Poring through the docs and issues, I found the `Charwidth` and related options. It seems like the following should cause everything to render in single width:
```
Charwidth=single
CharNarrowing=100
```

But now I get this:

![bomb-tiny](https://github.com/mintty/mintty/assets/499244/320f75d8-302c-42f7-a7b0-bbb7dbd9c10e)

With the glyph being scaled down. (That is, CharNarrowing doesn't seem to do anything.) I also tried `CharWidth=font`, which seems like I what I would want, but it didn't do anything and only appears in some of the documentation, so it may not be current.

I get the same result if I add

```
EmojiPlacement=full
```

I understand this stuff is really complicated, but I'm beginning to suspect a bug or at least a missing feature! I think this may have changed recently, since I just noticed it today.

**Comment by mintty** [comment]
U+1F4A3 is a double-width character by definition of Unicode.
Charwidth=single is not intended to make characters single-width but to adapt to terminals that would not support double-width,
therefore it just scales down if necessary; note it would also disable CJK double-width display; I'll add a warning to the manual.
You could use `printf '\e[2 Z\U1f4a3\e[0 Z'` (see https://github.com/mintty/mintty/wiki/CtrlSeqs#explicit-character-width) but it also zooms down the character. I'll consider whether your request to squeeze the character instead is a feature to be supported.

**Comment by BrianInglis** [comment]
Normal output looks fine to me at double width: 
```
$ printf '\U1f4a3\n'
💣
```
Just allow two spaces for double width chars, or suppress one following space, from your display. 
Sequence `\e[2 Z`...`\e[0 Z` in previous comment squishes the glyph, and trying sequence `\e[22 Z`...`\e[0 Z` zoom makes it unreadably tiny.

**Comment by tom7** [comment]
Right, I think the default behavior is good. My question is about fonts that have single-width glyphs for some codepoints.

It's reasonable to just say that this is invalid and unsupported. For what it's worth, I use this same font in other programs (e.g. emacs) where it does what I expect, probably because those are deferring to the font's metrics. Thanks for thinking about it. :)

**Comment by mintty** [comment]
Terminals do not (and cannot) render output according to font metrics.
They do not work well with proportional fonts anyway, usually you should use fixed-width fonts in a terminal.
The authoritative reference for character width in terminals is normally the wcwidth and wcswidth functions.
So you cannot assess the behaviour of a terminal by the behaviour of GUI programs.

**Comment by mintty** [comment]
> but it also zooms down the character. I'll consider whether your request to squeeze the character instead

I had misinterpreted the visual appearance of the emoji. In fact it is zoomed only horizontally (i.e. squeezed) which is the proper way to handle the situation I think. On the other hand, CJK characters are zoomed down as squeezing them would significantly impact their appearance.
